pennylane/estimator/resources_base.py

```diff
@@ -0,0 +1,412 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""Base class for storing resources."""
+from __future__ import annotations
+
+from collections import Counter, defaultdict
+from decimal import Decimal
+
+from .wires_manager import WireResourceManager
+
+DefaultGateSet = {
+    "X",
+    "Y",
+    "Z",
+    "Hadamard",
+    "CNOT",
+    "S",
+    "T",
+    "Toffoli",
+}
+
+
+class Resources:
+    r"""A container to track and update the resources used throughout a quantum circuit.
+    The resources tracked include number of gates, number of wires, and gate types.
+
+    Args:
+        wire_manager (:class:`~.pennylane.estimator.WireResourceManager`): A wire tracker class which contains the number of available
+            work wires, categorized as clean, dirty or algorithmic wires.
+        gate_types (dict): A dictionary storing operations (:class:`~.pennylane.estimator.ResourceOperator`) as keys and the number
+            of times they are used in the circuit (int) as values.
+
+    **Example**
+
+    The resources can be accessed as class attributes. Additionally, the :class:`~.pennylane.estimator.Resources`
+    instance can be nicely displayed in the console.
+
+    >>> H = plre.resource_rep(plre.Hadamard)
+    >>> X = plre.resource_rep(plre.X)
+    >>> Y = plre.resource_rep(plre.Y)
+    >>> wm = plre.WireResourceManager(work_wires=3)
+    >>> gt = defaultdict(int, {H: 10, X:7, Y:3})
+    >>>
+    >>> res = plre.Resources(wire_manager=wm, gate_types=gt)
+    >>> print(res)
+    --- Resources: ---
+     Total wires: 3
+        algorithmic wires: 0
+        allocated wires: 3
+             clean wires: 3
+             dirty wires: 0
+     Total gates : 20
+      'X': 7,
+      'Y': 3,
+      'Hadamard': 10
+
+    .. details::
+        :title: Usage Details
+
+        The :class:`~.pennylane.estimator.Resources` object supports arithmetic operations which allow for quick addition
+        and multiplication of resources. When combining resources, we can make a simplifying
+        assumption about how they are applied in a quantum circuit (in series or in parallel).
+
+        When assuming the circuits were executed in series, the number of algorithmic wires add
+        together. When assuming the circuits were executed in parallel, the maximum of each set of
```

Comment:
```suggestion
        together. When assuming the circuits were executed in series, the maximum of each set of
```
Assuming my previous comment was correct.